frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance/font_family_setting.dart

```diff
@@ -109,35 +111,61 @@ class _ThemeFontFamilySettingState extends State<ThemeFontFamilySetting> {
         .replaceAllMapped(camelCase, (m) => ' ${m.group(0)}');
   }
 
+  bool _isTextOverflowing(String text, TextStyle textStyle, double maxWidth) {
+    final textPainter = TextPainter(
+      text: TextSpan(text: text, style: textStyle),
+      maxLines: 1,
+      textDirection: ui.TextDirection.ltr,
+    )..layout(maxWidth: maxWidth);
+
+    return textPainter.didExceedMaxLines;
+  }
+
   Widget _fontFamilyItemButton(BuildContext context, TextStyle style) {
     final buttonFontFamily = parseFontFamilyName(style.fontFamily!);
-    return SizedBox(
-      key: UniqueKey(),
-      height: 32,
-      child: FlowyButton(
-        key: Key(buttonFontFamily),
-        onHover: (_) => FocusScope.of(context).unfocus(),
-        text: FlowyText.medium(
-          parseFontFamilyName(style.fontFamily!),
-          fontFamily: style.fontFamily!,
+
+    final bool isOverFlown = _isTextOverflowing(
+      buttonFontFamily,
+      Theme.of(context).textTheme.bodyMedium!.copyWith(
+            fontWeight: FontWeight.w500,
+            fontFamily: buttonFontFamily,
+          ),
+      110,
```

Comment:
Where does this "110" come from? Can we use a variable to replace it(like fontPreviewTextWidth?

Reply:
- It looks like the `AppFlowyPopover` has a  default max and min width of 160 and 80 resp. Here in this case, it looks like it  always keeps its max width constraint. So the complete width of the popover is taken as 160. 
- Then there is default margin for popover "6 each side", thus deducted 12 from 160. 
- Then came `FlowyButton`, it also has margins, "10 horizontally", thus can deduct 20 again. 
- This brings us to 128 as the available space for text. (almost)
- But still some texts was overflowing due to its fontsize, so I tried with smaller values, and 110 looked like a decent value to cover most of the text overflows. 

That was the process. 
What you think?  Is there any better approach that I can pick to figure this width?

Reply:
Got it. After playing around for a while, I am thinking is it better to add the tooltip on all the fonts? Because I noticed even though some font didn't overflow, it is still kind of hard to read it like the following one. 


<img width="238" alt="image" src="https://github.com/AppFlowy-IO/AppFlowy/assets/14248245/61be6630-c5fd-4ed6-84a4-da4f610735fe">

What do you think?  @MayurSMahajan @Xazin @AslamThachapalli @annieappflowy

Reply:
Ok. But won't this make the ui less smooth? Like the tooltip comes in the way of scrolling.

Preview Vid:

https://github.com/AppFlowy-IO/AppFlowy/assets/113328135/4974fcca-d3fc-4b28-ad18-12fbe1c2ed02

What do you think? Should I move forward with this?

Reply:
It looks fine to me though, I just feel the delay time of the tooltip makes it seem not to match the font that the current cursor points to. What do you think? Add a tooltip to **all the fonts** or just **the ones that overflow** is better? @Xazin @annieappflowy